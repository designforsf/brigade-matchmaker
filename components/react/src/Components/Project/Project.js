import React from 'react';

class Project extends React.Component {
  render() {
    let {
      attributes: { name, description, slackChannel, todoItems, progressItems,
        matchingConfig: { interests, skillsNeeded, skillsOffered }
      }
    } = this.props;
    return (
      <div className="project"> 
      <div className="projectCols">
        <div className="projectCol1">
          <h5 className="projectTitle">{name}</h5>
          <p>{description}</p>
          <p><strong>Contact Project</strong></p>
          <p><a href="#">{slackChannel}</a></p>
        </div>
        <div className="projectCol2">
          <h5 className="skillsHeader">Project Needs & Interests</h5>

          <p className="skillsCategory"><strong>Skills Needed:</strong></p>
          <div className="tagContainer">
            {skillsNeeded.map(skill =>
              <div className="tag">{skill}</div>
              )}
          </div>

          <p className="skillsCategory"><strong>Learning Opportunities:</strong></p>
          <div className="tagContainer">
            {skillsOffered.map(skill =>
              <div className="tag">{skill}</div>
            )}
          </div>

          <p className="skillsCategory"><strong>Civic Interests:</strong></p>
          <div className="tagContainer">
            {interests.map(interest =>
              <div className="tag">{interest}</div>
            )}
          </div>
      
        </div>
      </div>
        <details>
          <summary>Show project details</summary>
          <div className="detailsGrid">
            <ul className="details"><strong>Pending Tasks</strong></ul>
            <ul className="details"><strong>Progress Made</strong></ul>
            <div className="additional-info">
              <p className="additionalInfo"><strong>Additional Info</strong></p>
            </div>
          </div>
        </details>
      </div>
    );
  }
}

export default Project;

