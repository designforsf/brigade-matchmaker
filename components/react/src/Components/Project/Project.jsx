import React from 'react';
import Tag from '../Tag';

class Project extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
    };

    this.handleClick = this.handleClick.bind(this);
    this.renderTag = this.renderTag.bind(this);
  }

  handleClick(e) {
    const clickedName = e.target.textContent;
    this.setState({
      name: clickedName,
    });
  }

  renderTag(skill) {
    const defaultBorder = '1px solid #AA193A';
    const clickedBorder = '1px solid lightgreen';

    if (skill === this.state.name) {
      return <Tag name={skill} border={clickedBorder} onClick={this.handleClick} />;
    }
    return <Tag name={skill} border={defaultBorder} onClick={this.handleClick} />;
  }

  render() {
    const {
      attributes: {
        name, description, slackChannel, todoItems, progressItems, matchingConfig:
        { interests, skillsNeeded, skillsOffered },
      },
    } = this.props;
    return (
      <div className="project">
        <div className="projectCols">
          <div className="projectCol1">
            <h5 className="projectTitle">{name}</h5>
            <p>{description}</p>
            <p className="contact"><strong>Contact Project</strong></p>
            <p><button type="button">{slackChannel}</button></p>
          </div>
          <div className="projectCol2">
            <h5 className="skillsHeader">Project Needs & Interests</h5>

            <p className="skillsCategory"><strong>Skills Needed:</strong></p>
            <div className="tagContainer">
              { skillsNeeded.map(skill => this.renderTag(skill)) }
            </div>

            <p className="skillsCategory"><strong>Learning Opportunities:</strong></p>
            <div className="tagContainer">
              { skillsOffered.map(skill => this.renderTag(skill)) }
            </div>

            <p className="skillsCategory"><strong>Civic Interests:</strong></p>
            <div className="tagContainer">
              { interests.map(interest => this.renderTag(interest)) }
            </div>

          </div>
        </div>
        <details>
          <summary>Show project details</summary>
          <div className="detailsGrid">
            <ul className="details"><strong>Pending Tasks</strong></ul>
            <ul className="details"><strong>Progress Made</strong></ul>
            <div className="additional-info">
              <p className="additionalInfo"><strong>Additional Info</strong></p>
            </div>
          </div>
        </details>
      </div>
    );
  }
}

export default Project;
