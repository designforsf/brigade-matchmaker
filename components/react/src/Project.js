import React from 'react';

class Project extends React.Component {
  render() {
    let { title } = this.props;
    return (
      <div className="project" style={{border: "1px solid gray", padding: 16, margin: 16}}>
      <div className="projectCols" style={{border: "1px solid gray", display: "grid", gridGap: 10, gridTemplateColumns: "repeat(auto-fill, minmax(200px, 1fr))", padding: 16}}>
        <div className="projectCol1" style={{borderRight: "1px solid gray"}}>
          <h5>{title}</h5>
          <p>summary</p>
          <button>Contact Project Lead</button>
        </div>
        <div className="projectCol2">
          <h5>Project Needs & Interests</h5>
          <p><strong>Skills Needed:</strong></p>
          <p><strong>Learning Opportunities:</strong></p>
          <p><strong>Civic Interests:</strong></p>
        </div>
      </div>
        <details>
          <summary>Show project details</summary>
            <p>Pending Tasks</p>
            <p>Progress Made</p>
            <p>Additional Info</p>
        </details>
      </div>
    );
  }
}

export default Project;
