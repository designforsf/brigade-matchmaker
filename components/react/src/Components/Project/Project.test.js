import { shallow } from "enzyme";
import React from "react";
import Project from ".";

describe("Project", () => {
  let project;
  beforeEach(() => (project = shallow(<Project />)));
  it("matches the snapshot", () => {
    expect(project).toMatchSnapshot();
  });
});
