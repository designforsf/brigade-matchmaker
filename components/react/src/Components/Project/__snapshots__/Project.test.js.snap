// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project matches the snapshot 1`] = `undefined`;
